docs/proving-authentication.mdx

```diff
@@ -0,0 +1,189 @@
+backend must follow in using this data are specific.
+
+- First your application must determine wether a user with the `address`
+  provided in the `account-proof` data structure has authenticated with your
+  application before.
+    - If a user with the `address` has authenticated with your application
+      before, then your system must query the `timestamp` associated with the
+      previous authentication for this user.
+        - If the `timestamp` is less than or equal to the `timestamp` associated
+          with the users previous authentication, your system must abort and
+          reject the authentication attempt.
+        - If the `timestamp` is greater than the `timestamp` associated with the
+          users previous authentication, then continue.
+    - If a user with the `address` _has not_ authenticated with your application
+      before, then continue.
+
+- Your application must determine whether the `CompositeSignture` in the
+  `account-proof` data structure contains a valid signature. Your application
+  will do this by:
+    - Reconstructing the data the was signed to produce the
+      `CompositeSignature`. Pseudocode:
+
+      ```javascript
+      import rlp from "@onflow/rlp"
+
+      const Address = AccountProof.address
+      const Timestamp = AccountProof.timestamp
+      const Message = AccountProof.message
+
+      const rightPaddedHexBuffer = (value, pad) =>
+        Buffer.from(value.padEnd(pad * 2, 0), "hex")
+
+      const YourApplicationsDomainTag = "Awesome-Dapp"
+      const DomainTag = 
+        rightPaddedHexBuffer(
+            Buffer.from("FLOW-V0.0-user" + YourApplicationsDomainTag).toString("hex"), 
+            32
+        ).toString("hex")
+
+      const Data = DomainTag + rlp([ 
+        Address,
+        Timestamp,
+        Message
+      ])
+      ```
+
+  - Verify that the `account-proof` signature is valid for the reconstructed
+    data. Your application can do this through executing it's own Cadence
+    script, or using `fcl.verifyUserSignature`:
+
+```javascript
+...
+const isValid = await fcl.verifyUserSignature(
+    Data,
+    AccountProof.signatures,
+    DomainTag
+)
+```
+
+- If the signature has been proved valid, your application has then successfully
+  proved that the user does control the account with address they authenticated
+  with.
+- Your application must finally save the `AccountProof.timestamp` associated
+  with this authentication attempt alongside the `address` for the user for
+  future reference.
+
+### Proving User Authentication using `user-signature`
+
+If your user's wallet does not return the the FCL `account-proof` service, your
+application can still prove user authentication using the `user-signature`
+service, as long as it is supported by your user's wallet.
+
+Your application's backend must first generate a random message that you wish
+for your user to sign, with their wallet, using the `user-signature` service.
```

Comment:
Your application's backend must first generate a random message for the user to sign with their wallet, using the `user-signature` service.